### ISSUE TITLE
Expose API in a library

### ISSUE THREAD

**Issue by Zeex** [Southclaws/pawn#244]
I just thought that some compiler functions/types could be exposed as a library similar to how clang does it with its libclang, so that it would be possible to create new tools without reimplementing the compiler. For example, it could be used to implement auto-completion in editors or maybe linters.

What do you think?

There is libpawnc already but it provides only a couple of functions for embedding Pawn in other applications; ^ this would provide access to internal objects like global and local symbols (functions, constants, etc).

Though I have a feeling that this may be extremely hard to do (and/or pointless), probably harder than writing a parser from scratch...

**Comment by Southclaws** [comment]
This would be interesting, my only concern for editor tools is portability - C/++ isn't the easiest language to make portable! Either way though, it would be cool - depending on the amount of work required...

Related: I am working on a [parser for Pawn](https://github.com/Southclaws/pawn-parser) in the Go language which provides a syntax tree for a given script which I'm planning to use for static analysis tools.

**Comment by Zeex** [comment]
That looks interesting. 

C is pretty portable, it might be even the most portable language (in terms of supported OSes), and a lot of languages can interface with C libraries. But yeah, it's a pain in the ass to work with.

**Comment by Southclaws** [comment]
Yeah maybe portability wasn't the right choice of wort there, pain in the ass to work with is more descriptive!
